config/routes.rb

```diff
@@ -2,9 +2,15 @@
   root 'homes#index'
   devise_for :users
   resources :users
-  resources :recommendations
   resources :searches, only: [:index, :show, :destroy]
   resources :tweets
+  resources :recommendations
+  resources :recommendations, only: [:show] do
+      member do
+        put "like", to: "recommendations#upvote", defaults: { format: 'json' }
+        put "dislike", to: "recommendations#downvote", defaults: { format: 'json' }
+      end
+    end
```

Comment:
`end` at 13, 4 is not aligned with `resources :recommendations, only: [:show] do` at 8, 2